react/features/virtual-background/components/VirtualBackgroundPreview.js

```diff
@@ -0,0 +1,270 @@
+import { createLocalTracksF } from '../../base/tracks/functions';
+import { createVirtualBackgroundEffect } from '../../stream-effects/virtual-background';
+import { toggleBackgroundEffect } from '../actions';
+
+const videoClassName = 'video-preview-video flipVideoX';
+
+/**
+ * The type of the React {@code Component} props of {@link VirtualBackgroundPreview}.
+ */
+export type Props = {
+
+    /**
+     * The deviceId of the camera device currently being used.
+     */
+    _currentCameraDeviceId: string,
+
+    /**
+     * The virtual background object.
+     */
+
+    _virtualBackground: Object,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function,
+
+    /**
+     * Represents the virtual background setted options.
+     */
+    options: Object,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * The type of the React {@code Component} state of {@link VirtualBackgroundPreview}.
+ */
+type State = {
+
+    /**
+     * Loader activated on setting virtual background.
+     */
+    loading: boolean,
+
+    /**
+     * An array of all the jitsiTracks and eventual errors.
+     */
+    trackData: Object,
+
+    /**
+     * Activate the selected device camera only.
+     */
+    selectedTrack: Object
+};
+
+/**
+ * Implements a React {@link Component} which displays the virtual
+ * background preview.
+ *
+ * @extends Component
+ */
+class VirtualBackgroundPreview extends Component<Props, State> {
+    _componentWasUnmounted: boolean;
+
+    /**
+     * Initializes a new {@code VirtualBackgroundPreview} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            trackData: {},
+            loading: false,
+            selectedTrack: null
+        };
+    }
+
+    /**
+     * Creates and updates the track data.
+     *
+     * @returns {void}
+     */
+    async _setTracks() {
+        this._disposeTracks(this.state.trackData);
+
+        const [ trackData ] = await createLocalTracksF({
+            cameraDeviceId: this.props._currentCameraDeviceId,
+            devices: [ 'video' ]
+        });
+
+
+        // In case the component gets unmounted before the tracks are created
+        // avoid a leak by not setting the state
+        if (this._componentWasUnmounted) {
+            this._disposeTracks(trackData);
+        } else {
+            this.setState({
+                trackData
+            });
+        }
+
+        // background preview should open the selected device camera only
+        const selectedTrack = this.state.trackData;
+
+        this.setState({
+            selectedTrack
+        });
+        if (this.props._virtualBackground?.backgroundEffectEnabled && this.state.selectedTrack) {
+            await this.state.selectedTrack.setEffect(
```

Comment:
If we are using `createLocalTracksF` there is no need to apply the effect ourselves, it should've been applied already.